src/ts/background/services/account-balance.ts

```diff
@@ -0,0 +1,29 @@
+import { ApiPromise } from '@polkadot/api'
+import { WsProvider } from '@polkadot/rpc-provider'
+import { Network } from '../../constants/networks'
+import { ChainProperties } from '@polkadot/types'
+import { formatBalance } from '@polkadot/util'
+
+export async function getBalance (
+  address: string,
+  network: Network,
+  withFormat?: boolean
+): Promise<Number | string> {
+  const provider = new WsProvider(network.rpcServer)
+  try {
+    const api = await ApiPromise.create(provider)
+    const balance = await api.query.balances.freeBalance(address)
+    if (!withFormat) return Number(balance.toString())
+    const properties = await api.rpc.system.properties()
+    const chainProperties = (properties as ChainProperties)
+    formatBalance.setDefaults({
+      decimals: chainProperties.tokenDecimals,
+      unit: chainProperties.tokenSymbol
+    })
+    return formatBalance(balance.toString())
+  } catch (e) {
+    return 'N/A'
```

Comment:
I think we probably want service to throw an error with a message and let caller decide how to display it accordingly.

Reply:
Yep, updated

> I think we probably want service to throw an error with a message and let caller decide how to display it accordingly.

Reply:
Throw new Error

Reply:
Moved it from background/services to services